hibernate-core/src/main/java/org/hibernate/tuple/PropertyFactory.java

```diff
@@ -148,14 +150,16 @@ public static VersionProperty buildVersionProperty(
 	 * @param property The mapped property.
 	 * @param lazyAvailable Is property lazy loading currently available.
 	 *
+	 * @param creationContext
 	 * @return The appropriate NonIdentifierProperty definition.
 	 */
 	public static NonIdentifierAttribute buildEntityBasedAttribute(
 			EntityPersister persister,
 			SessionFactoryImplementor sessionFactory,
 			int attributeNumber,
 			Property property,
-			boolean lazyAvailable) {
+			boolean lazyAvailable,
+			PersisterCreationContext creationContext) {
```

Comment:
I'm assuming that changing this SPI is OK in master.

Note to self: when this gets backported, the old version of the method should be deprecated.

Reply:
Same comment as previous